lhotse/recipes/himia.py

```diff
@@ -0,0 +1,252 @@
+    return recording_set, supervision_set
+
+
+def _prepare_cw_test(corpus_path: Path) -> Tuple[RecordingSet, SupervisionSet]:
+    """
+    Returns the RecodingSet and SupervisionSet of test dataset.
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :return: the RecodingSet and SupervisionSet of test dataset.
+    """
+    recordings = []
+    supervisions = []
+    cw_test_path = corpus_path / "cw_test/16k_wav_file"
+    logging.info(f"Searching wav files in {cw_test_path}")
+    transcript_path = corpus_path / "cw_test/resource/transcription.txt"
+    transcript_dict = {}
+    with open(transcript_path, "r", encoding="utf-8") as f:
+        for line in f.readlines():
+            # Examples of line:
+            # 0001_M02_01_fast_0001.wav▸你好米
+            # 0001_M02_01_fast_0002.wav▸你好你好
+            # 0001_M02_01_fast_0003.wav▸你好亚
+
+            idx_transcript = line.split()
+            transcript = " ".join(idx_transcript[1:])
+            transcript_dict[idx_transcript[0]] = transcript
+
+    assert len(transcript_dict) == _TOTAL_NUM_WAVS["cw_test"]
+
+    for wav_name in tqdm(transcript_dict, total=_TOTAL_NUM_WAVS["cw_test"]):
+        audio_path = cw_test_path / wav_name
+        audio_path = audio_path.resolve()
+        text = transcript_dict[wav_name]
+        assert audio_path.is_file(), f"{audio_path} does not exist."
+        recording = Recording.from_file(audio_path)
+        recordings.append(recording)
+        audio_file_name = audio_path.stem
+        speaker = audio_file_name.split("_")[0]
+        segment = SupervisionSegment(
+            id=audio_file_name,
+            recording_id=audio_file_name,
+            start=0.0,
+            duration=recording.duration,
+            channel=0,
+            language="Chinese",
+            speaker=speaker,
+            text=text.strip(),
+        )
+        supervisions.append(segment)
+
+    recording_set = RecordingSet.from_recordings(recordings)
+    supervision_set = SupervisionSet.from_segments(supervisions)
+    validate_recordings_and_supervisions(recording_set, supervision_set)
+    return recording_set, supervision_set
+
+
+def prepare_himia(
+    corpus_dir: Pathlike,
+    output_dir: Optional[Pathlike] = None,
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    Returns the manifests which consist of the Recordings and Supervisions
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :return: a Dict whose key is the dataset part, and the value is Dicts with the keys 'recordings' and 'supervisions'.
+    """
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+    manifests = defaultdict(dict)
+    train_dev_dataset_parts = ["train", "dev", "test", "cw_test"]
```

Comment:
You might want to follow how eg LibriSpeech recipe does auto-discovery of the available parts in case somebody only needs a subset.

Reply:
Updated according to your comments. Hope for your further reviews.